staging/src/k8s.io/client-go/rest/with_retry.go

```diff
@@ -245,6 +285,54 @@ func (r *withRetry) After(ctx context.Context, request *Request, resp *http.Resp
 	}
 }
 
+func (r *withRetry) WrapPreviousError(currentErr error) error {
+	if currentErr == nil || r.previousErr == nil {
+		return currentErr
+	}
+	if !isContextError(currentErr) {
```

Comment:
is `Get "http://foo.bar/api/v1": context deadline exceeded` a contextError or a URL error?

Reply:
both `errors.Is(err, &url.Error{})` and `errors.Is(err, context.DeadlineExceeded)` will return true. This PR preserves this behavior for an `err` that is wrapped by `WrapPreviousError`.